### PR TITLE
KAFKA-2599 Metadata#getClusterForCurrentTopics can throw NPE even wit…

### PR DESCRIPTION
**Pull request by eribeiro**
…h null checking

### CODE REVIEW

**Comment by eribeiro** [comment]
Well, the failed test (kafka.consumer.MetricsTest > testMetricsLeak FAILED) looks unrelated to the patch.

**Comment by eribeiro** [comment]
Also, I should have made this issue a MINOR. Excuse me.

File: clients/src/main/java/org/apache/kafka/clients/Metadata.java [hunk 1]
```diff
@@ -237,11 +238,13 @@ public void removeListener(Listener listener) {
 
     private Cluster getClusterForCurrentTopics(Cluster cluster) {
         Collection<PartitionInfo> partitionInfos = new ArrayList<>();
+        List<Node> nodes = Cluster.empty().nodes();
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/Metadata.java]: This seems a roundabout way of saying Collections.emptyList, no?

Comment (eribeiro) [inline clients/src/main/java/org/apache/kafka/clients/Metadata.java]: Yeah, you right. I realize that yet yesterday after pushing the repo (shrug). Gonna fix that.

**Comment by ijuma** [comment]
LGTM

**Comment by guozhangwang** [comment]
LGTM.

**Pull request closed without merging**